### DISCUSSION TITLE
How ethers connects to wallet when using wallet connect

### DISCUSSION THREAD

**Discussion by Agilulfo1820** [WalletConnect/walletconnect-monorepo#2436] [Developer Support] [answered]
When using wallet connect I see that a provider is created and passed to ethers/web3, after that each action I do (sending Eth, signing message, etc.) prompts inside the wallet I used to connect. How does ethers know which wallet connect client and session to use to communicate with the wallet? As far as I know the provider that is initialized it's just the RPC URL and the chain Id. Is  it something built inside ethers.js? How does this work?

I'm trying to build this on another non evm chain that doesn't support ethers but I'm struggling to understand this aspect.

I'm using wallet connect V2. SignClient on the web and Web3Wallet inside the wallet. I am able to send requests and process them, but for now I have to do an if/else inside my code and execute wallet connect requests (if connected with WC) or the blockchain SDK if connected with the extension. I would like to update the SDK in a way that I don't have to do this check (as it happens on Ethereum).

**Comment by finessevanes** [comment, marked as answer]
When you establish a connection with a wallet using WalletConnect, it initiates a session identified by a unique session ID. This session ID facilitates the communication between the dapp and the wallet. The providers for ethers.js or web3.js, like ethereum-provider or universal-provider, are initialized with the RPC URL and the chain ID. However, these libraries (ethers.js and web3.js) essentially serve as wrappers around our providers.

They play a crucial role in formatting transaction payloads before calling our provider's `.request` function. Once the `.request` function is triggered, it's up to the WalletConnect provider to route the request to the appropriate session based on the session ID. Hence, the WalletConnect provider is responsible for managing the interactions between the dapp and the wallet.
